input/kafkamdm/kafkamdm.go

```diff
@@ -231,12 +231,22 @@ func (k *KafkaMdm) consumePartition(topic string, partition int32, currentOffset
 	partitionOffsetMetric := partitionOffset[partition]
 	partitionLogSizeMetric := partitionLogSize[partition]
 	partitionLagMetric := partitionLag[partition]
+	var initialOffset int64
 	offset, err := k.client.GetOffset(topic, partition, sarama.OffsetNewest)
 	if err != nil {
 		log.Error(3, "kafka-mdm failed to get log-size of partition %s:%d. %s", topic, partition, err)
 	} else {
 		partitionLogSizeMetric.Set(int(offset))
 	}
+	if currentOffset == sarama.OffsetNewest {
+		initialOffset = offset
+	} else if currentOffset == sarama.OffsetOldest {
+		initialOffset, err = k.client.GetOffset(topic, partition, currentOffset)
+		if err != nil {
+			log.Error(3, "kafka-mdm failed to get oldest offset of partition %s:%d. %s", topic, partition, err)
+		}
+	}
+
 	if currentOffset >= 0 {
 		// we cant set the offsetMetrics until we know what offset we are at.
```

Comment:
It looks like this statement is no longer true.  We do know the offset we are at, it is stored in initialOffset